spec/acceptance/docker_full_spec.rb

```diff
@@ -0,0 +1,643 @@
+            }
+      EOS
+      apply_manifest(pp, :catch_failures => true)
+  end
+end
+
+describe 'docker full acceptance tests section 1 - testing class {\'docker\':}' do
+  context 'test 1.1 - Install and configure Docker and Docker Daemon' do
+    it 'applies the manifest' do
+      pp=<<-EOS
+        class { 'docker': }
+      EOS
+
+      apply_manifest(pp, :catch_failures => true)
+      unless fact('selinux') == 'true'
+        apply_manifest(pp, :catch_changes => true)
+      end
+    end
+
+    it ': ps -aux should be running docker process' do
+      shell('ps -aux | grep docker') do |r|
+        expect(r.stdout).to match(/\/usr\/bin\/docker/)
+      end
+    end
+    it ': docker ps should output table headers' do
+      shell('docker ps') do |r|
+        expect(r.stdout).to match(/CONTAINER ID\s+IMAGE\s+COMMAND\s+CREATED\s+STATUS\s+PORTS\s+NAMES/)
+      end
+    end
+  end 
+
+  context 'test 1.3a - Bound to a particular TCP socket' do
+    it 'applies the manifest' do
+      pp=<<-EOS
+        class { 'docker': 
+          tcp_bind    => 'tcp://127.0.0.1:4444',
+        }
+      EOS
+
+      apply_manifest(pp, :catch_failures => true)
+      unless fact('selinux') == 'true'
+        apply_manifest(pp, :catch_changes => true)
+      end
+    end
+
+    it ': netstat should show docker listening' do
+      shell('netstat -tulpn | grep docker') do |r|
+        expect(r.stdout).to match(/tcp\s+0\s+0\s+127.0.0.1:4444\s+0.0.0.0\:\*\s+LISTEN\s+\d+\/docker/)
+      end
+    end
+    it ': docker ps should output the table headers' do
+      shell('docker ps') do |r|
+        expect(r.stdout).to match(/CONTAINER ID\s+IMAGE\s+COMMAND\s+CREATED\s+STATUS\s+PORTS\s+NAMES/)
+      end
+    end
+  end  
+
+  context 'test 1.3b - Bound to a particular unix socket' do
+    it 'applies the manifest' do
+      pp=<<-EOS
+        class { 'docker': 
+          socket_bind => 'unix:///var/run/docker.sock',
+        }
+      EOS
+
+      apply_manifest(pp, :catch_failures => true)
+      unless fact('selinux') == 'true'
+        apply_manifest(pp, :catch_changes => true)
+      end
+    end
+
+    it ': ps -aux should show docker listening on a unix socket' do
+      shell('ps -aux | grep docker') do |r|
+        expect(r.stdout).to match(/unix:\/\/\/var\/run\/docker.sock/)
+      end
+    end
+    it ': docker ps should output the table headers' do
+      shell('docker ps') do |r|
+        expect(r.stdout).to match(/CONTAINER ID\s+IMAGE\s+COMMAND\s+CREATED\s+STATUS\s+PORTS\s+NAMES/)
+      end
+    end
+  end
+end
+
+describe 'docker full acceptance tests section 2 - testing docker::image' do
+  before(:each) do
+    # Delete all existing images
+    shell('docker rmi $(docker images -q) || true')
+    # Check to make sure no images are present
+    shell('docker images | wc -l') do |r|
+      expect(r.stdout).to match(/^0|1$/)
+    end
+  end
+
+  context 'test 2.1 - Download Image' do
+    it 'runs test' do
```

Comment:
describe the test. So in full this should probably be:

    docker::image should successfully download an image from the Docker Hub

You don't actually need the context block here.